src/hooks/useTimelineUrlState.ts

```diff
@@ -19,61 +20,62 @@ const defaultState: TimelineState = {
 };
 
 export function useTimelineUrlState() {
-  const [searchParams, setSearchParams] = useSearchParams();
+  const search = useSearch();
+  const navigate = useNavigate();
 
   const getStateFromUrl = useCallback((): TimelineState => {
     return {
-      timelineView:
-        (searchParams.get("view") as TimelineView) || defaultState.timelineView,
-      selectedDay: searchParams.get("day") || defaultState.selectedDay,
-      selectedTime:
-        (searchParams.get("time") as TimeFilter) || defaultState.selectedTime,
+      timelineView: search.view || defaultState.timelineView,
+      selectedDay: search.day || defaultState.selectedDay,
+      selectedTime: search.time || defaultState.selectedTime,
       selectedStages:
-        searchParams.get("stages")?.split(",").filter(Boolean) ||
+        search.stages?.split(",").filter(Boolean) ||
```

Comment:
defaults can be set on the router level.
also search.stages should be an array of strings